### ISSUE TITLE
LibSass `str-slice()` returning wrong value compared to Dart and Ruby Sass

### ISSUE THREAD

**Issue by oscarmarcelo** [sass/libsass#2808]
LibSass `str-slice()` returns the wrong result compared to Dart and Ruby Sass when using `$start-at` with a negative number bigger than `$string` length.

## input.scss

```scss
test {
  content: str-slice(abcdef, -10, 2)
}
```

## Actual results

libsass 3.5.2
```css
test {
  content: abc; }
```

## Expected result

dart sass 1.6.2
```css
test {
  content: ab; }
```

ruby sass 3.5.6
```css
test {
  content: ab; }
```

**Comment by mgreter** [comment]
http://libsass.ocbnet.ch/srcmap/#dGVzdCB7CiAgY29udGVudDogc3RyLXNsaWNlKGFiY2RlZiwgLTEwLCAyKQp9

**Comment by nschonni** [comment]
This should be fixed in #2947
